docs/content/tutorial/step_04.ngdoc

```diff
@@ -1,199 +1,310 @@
+difficult to navigate and find the code we are looking for.
 
-```html
-  Search: <input ng-model="query">
-  Sort by:
-  <select ng-model="orderProp">
-    <option value="name">Alphabetical</option>
-    <option value="age">Newest</option>
-  </select>
+Instead we should put each feature/entity in its own file. Each stand-alone controller will be
+defined in its own file, each component will be defined in each own file, etc.
 
+Luckily, we don't need to change anything with respect to that guideline in our code, since we have
+already defined our `phoneList` component in its own `phone-list.component.js` file. Good job!
+
+We will keep this in mind though, as we add more features.
 
-  <ul class="phones">
-    <li ng-repeat="phone in phones | filter:query | orderBy:orderProp">
-      <span>{{phone.name}}</span>
-      <p>{{phone.snippet}}</p>
-    </li>
-  </ul>
-```
 
-We made the following changes to the `index.html` template:
+## Organizing by Feature
 
-* First, we added a `<select>` html element named `orderProp`, so that our users can pick from the
-two provided sorting options.
+So, now that we learned we should put everything in its own file, our `app/` directory will soon be
+full with dozens of files and specs (remember we keep our unit test files next to the corresponding
+source code files). What's more important, logically related files will not be groupped together; it
+will be really difficult of locate all files related to a specific section of the application and
+make a change or fix a bug.
 
-<img class="diagram" src="img/tutorial/tutorial_04.png">
+So, what shall we do?
 
-* We then chained the `filter` filter with {@link ng.filter:orderBy `orderBy`}
-filter to further process the input into the repeater. `orderBy` is a filter that takes an input
-array, copies it and reorders the copy which is then returned.
+Well, we are going to group our files into directories _by feature_. For example, since we have a
+section in our application that lists phones, we will put all related files into a `phone-list/`
+directory under `app/`. We are soon to find out that certain features are used across different
+parts of the application. We will put those inside `app/core/`.
 
-Angular creates a two way data-binding between the select element and the `orderProp` model.
-`orderProp` is then used as the input for the `orderBy` filter.
+<div class="alert alert-info">
+  <p>
+    Other typical names for our `core` directory are `common` and `components`. The latter is kind
+    of misleading though, as it will contain other things than components as well.
+  </p>
+  <p>
+    (This is mostly a relic of the past, when "components" just meant the generic building blocks of
+     an application.)
+  </p>
+</div>
+
+Based on what we have discussed so far, here is our directory/file layout for the `phoneList`
+"feature":
+
+```
+  app/
+    phone-list/
+      phone-list.component.js
+      phone-list.component.spec.js
+    app.js
+```
```

Comment:
In this listings (there are more below), I show the current structure after a specific refactoring step (e.g. move feature into own directory, make it its own module, use external template etc).
But even in the final structure shown below, there is no `core/` directory yet - it is createdin the next step, when we create a "shared" feature (the checkmark filter).

I think it would be confusing to show a directory (or folder :stuck_out_tongue:) that we haven't created yet, or to create a directory that we don't need. I can still add it if you feel strongly about it :stuck_out_tongue: